internal/clients/github/client.go

```diff
@@ -511,3 +513,54 @@ func newHttpClients(ctx context.Context, token string) (client *http.Client, gra
 
 	return clientWithSecondaryRateLimit, clientWithAcceptHeader, nil
 }
+
+var enterpriseQuery struct {
+	Enterprise struct {
+		OwnerInfo struct {
+			MembersCanChangeRepositoryVisibilitySetting string
+		}
+		Name          string
+		Url           string
+		Id            string
+		DatabaseId    int64
+		ViewerIsAdmin bool
+	} `graphql:"enterprise(slug: $slug)"`
+}
+
+func (c *Client) CollectEnterprises() ([]githubcollected.Enterprise, error) {
+	if len(c.enterprises) == 0 {
+		return nil, nil
+	}
+	enterprises, err := c.collectSpecificEnterprises()
+	if err != nil {
+		return nil, err
+	}
+
+	return enterprises, nil
+}
+
+func (c *Client) collectSpecificEnterprises() ([]githubcollected.Enterprise, error) {
+	res := make([]githubcollected.Enterprise, 0, len(c.enterprises))
+
+	for _, enterprise := range c.enterprises {
+
+		variables := map[string]interface{}{
+			"slug": githubv4.String(githubv4.String(enterprise)),
+		}
+
+		err := c.GraphQLClient().Query(c.context, &enterpriseQuery, variables)
```

Comment:
skip the failing enterprise with informative error instead of failing the function